src/ValueHandler/AttributeValueHandler.php

```diff
@@ -156,17 +156,35 @@ private function getAttributeValue(AttributeInterface $attribute, $value)
         if ($attribute->getType() === SelectAttributeType::TYPE) {
             $attributeConfiguration = $attribute->getConfiguration();
             $possibleOptionsCodes = array_map('strval', array_keys($attributeConfiguration['choices']));
-            if (!in_array($value, $possibleOptionsCodes, true)) {
-                throw new \InvalidArgumentException(
-                    sprintf(
-                        'This select attribute can only save existing attribute options. ' .
-                        'Attribute option with the given %s code does not exist.',
-                        $value
-                    )
-                );
-            }
 
-            return [$value];
+            if (is_array($value)) {
+                // multi select
+                $invalid = array_diff($value, $possibleOptionsCodes);
+                if (!empty($invalid)) {
+                    throw new \InvalidArgumentException(
+                        sprintf(
+                            'This select attribute can only save existing attribute options. ' .
+                            'Attribute option codes [%s] do not exist.',
+                            implode(', ', $invalid)
+                        )
+                    );
+                }
+
+                return $value;
+
+            } else {
+                if (!in_array($value, $possibleOptionsCodes, true)) {
+                    throw new \InvalidArgumentException(
+                        sprintf(
+                            'This select attribute can only save existing attribute options. ' .
+                            'Attribute option with the given %s code does not exist.',
+                            $value
+                        )
+                    );
+                }
+
+                return [$value];
+            }
```

Comment:
This else statement it's not necessary and should be removed.

Reply:
I mean only the else statement itself and not its content. So it should become:

```php
            if (is_array($value)) {
                // multi select
                $invalid = array_diff($value, $possibleOptionsCodes);
                if (!empty($invalid)) {
                    throw new \InvalidArgumentException(
                        sprintf(
                            'This select attribute can only save existing attribute options. ' .
                            'Attribute option codes [%s] do not exist.',
                            implode(', ', $invalid)
                        )
                    );
                }

                return $value;

            } 
            if (!in_array($value, $possibleOptionsCodes, true)) {
                throw new \InvalidArgumentException(
                    sprintf(
                        'This select attribute can only save existing attribute options. ' .
                        'Attribute option with the given %s code does not exist.',
                        $value
                    )
                );
            }

            return [$value];
```

Reply:
Sure, can do that.

I just thought it reduces the cognitive load glancing over the code. You need read to the end of the previous `if` to know it will never fall through. The `else` immediately tells you that this is the other branch and will never be executed if `is_array` was true.

Reply:
Maybe a better refactor is:

```php
                $value = (array)$value;
                $invalid = array_diff($value, $possibleOptionsCodes);
                if (!empty($invalid)) {
                    throw new \InvalidArgumentException(
                        sprintf(
                            'Cannot save option codes [%s] on the "%s" select attribute. ' .
                            implode(', ', $invalid),
                            $attribute->getCode()
                        )
                    );
                }

                return $value;
```

Reply:
The current PR handles multi select and select differently, because Akeneo handles them differently.

I don't have an opinion either way. If you want to handle a select field no matter what type it is in Akeneo, you proposal would work as well.